crates/rune-modules/src/http.rs

```diff
@@ -241,6 +248,20 @@ impl RequestBuilder {
         }
     }
 
+    #[rune::function]
+    fn basic_auth(self, username: String, password: Option<String>) -> Self {
```

Comment:
`Option<String>` was used here because attempting to use `Option<&str>` results in a trait not impl error.

Reply:
You can use `Option<Ref<str>>` to avoid cloning the string.

Reply:
Oh, I always forget about `Ref<T>`